### PR TITLE
Gps data capturer enhance

### PR DESCRIPTION
**Pull request by lynnliu655**
# Description
Enhance the  GpsDataCapturer code and clean GpsDataAnalyzer files. 
## Type of change
 * Simplify the test automation process as below: 
**Start Capture**
```adb shell am start -a com.google.gpsdatacapturer.START_CAPTURE --ez fused_location_type false --ez batching false ```
**Stop Capture**
```adb shell am startservice -a com.google.gpsdatacapturer.STOP_CAPTURE```

 * Add number of satellites used in fix on UI.
 * Add  number of satellites used in fix and average of top 4 satellites signal in the xml file. 

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java [hunk 1]
```diff
@@ -26,15 +26,16 @@ public void onLocationResult(LocationResult locationResult) {
...
     }
 
-    @Override
-    public void onGpsStatusChanged(int event) {
-        gpsDataCaptureService.onGpsStatusChanged(event);
-    }
+//    @Override
+//    public void onGpsStatusChanged(int event) {
+//        gpsDataCaptureService.onGpsStatusChanged(event, -1);
+//    }
```
Comment (lynnliu655) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java]: Will remove line 37-40

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java [hunk 2]
```diff
@@ -98,30 +95,24 @@ public IBinder onBind(Intent intent) {
...
-            }
+//
+//            //Extra the location api type
+//            boolean type_from_intent = intent.getBooleanExtra("fused_location_type", false);
+//            Log.d(TAG, "type_from_intent: " + type_from_intent);
+//            LocationApiType type = LocationApiType.LOCATIONMANAGER;
+//            if (type_from_intent) {
+//                type = LocationApiType.FUSEDLOCATIONPROVIDERCLIENT;
+//            }
+//            Log.d(TAG, "LocationApiType: " + type);
```
Comment (lynnliu655) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java]: Will remove line 98-106

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 3]
```diff
@@ -90,11 +97,15 @@ protected void onCreate(Bundle savedInstanceState) {
         //Start the service
         startAndBindGpsDataCaptureService();
 
+        //get the called intent
+        adbIntent = this.getIntent();
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: You may want to rename it to a generic name such as launch intent

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 4]
```diff
@@ -196,6 +208,29 @@ public void onServiceConnected(ComponentName name, IBinder service) {
             //get the gpsDataCaptureService
             gpsDataCaptureService =
                     ((GpsDataCaptureService.GpsDataCaptureBinder) service).getService();
+
+            //start capture via intent
+            if(adbIntent != null) {
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: You may want to compare it with action string "com.google.gpsdatacapturer.START_CAPTURE"

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 5]
```diff
@@ -196,6 +208,29 @@ public void onServiceConnected(ComponentName name, IBinder service) {
...
+                //Extra the location api type
+                if(adbIntent.getExtras() != null){
+                    Log.d(TAG, "Intent extra is " + adbIntent.getExtras().toString());
+                    boolean type_from_intent = adbIntent.getBooleanExtra("fused_location_type", false);
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: this extra would be present only for action com.google.gpsdatacapturer.START_CAPTURE
Also the user may forget to add it in the command line. Please make sure to add check for missing extra

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 6]
```diff
@@ -196,6 +208,29 @@ public void onServiceConnected(ComponentName name, IBinder service) {
...
+                    Log.d(TAG, "fused_location_type: " + type_from_intent);
+                    if(type_from_intent){
+                        locationApiType = LocationApiType.FUSEDLOCATIONPROVIDERCLIENT;
+                    }
+                    Log.d(TAG, "LocationApiType: " + locationApiType);
+                }
+                //Start capture if the action is START_CAPTURE
+                if (adbIntent.getAction() != null) {
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: You may want to compare it with action string "com.google.gpsdatacapturer.START_CAPTURE"

File: GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -9,16 +9,19 @@
...
 
     <application>
         <!-- SERVICES-->
         <service android:name="com.google.sharedlibrary.service.GpsDataCaptureService"
             android:exported="true"
             android:enabled="true">
             <intent-filter>
-                <action android:name="com.google.sharedlibrary.service.GpsDataCaptureService.STARTCAPTURE"/>
-                <action android:name="com.google.sharedlibrary.service.GpsDataCaptureService.STOPCAPTURE"/>
+                <action android:name="com.google.gpsdatacapturer.STOP_CAPTURE"/>
```
Comment (AdsuleA) [inline GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml:20]: In a future patch, is it possible to move handling of intent with action com.google.gpsdatacapturer.STOP_CAPTURE to activity?

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/LocationManagerListener.java [hunk 8]
```diff
@@ -24,7 +34,53 @@ public LocationManagerListener(GpsDataCaptureService service) {
...
+                averageTop4SignalStrength /= size;
+            }
+
+            Log.d(TAG, "Satellites visible: " + satellitesVisible);
+            Log.d(TAG, "Satellites used in fix: " + satellitesUsedInFix);
+            Log.d(TAG, "Average of satellites used in fix signal strength: " + averageSignalStrength);
+            Log.d(TAG, "Average of top 4 strongest signal strength: " + averageTop4SignalStrength);
+
+            gpsDataCaptureservice.setAverageOfTop4Signal(averageTop4SignalStrength);
+        }
```
Comment (AdsuleA) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/LocationManagerListener.java]: could we move this processing to onGpsStatusChanged method in gpsDataCaptureservice

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java [hunk 9]
```diff
@@ -177,10 +167,8 @@ public void onLowMemory() {
 
     /**
      * Start capturing data from GPS via the chosen location api
-     *
-     * @param locationApiType the chosen location api
      */
-    public void startCapture(LocationApiType locationApiType) {
```
Comment (AdsuleA) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java:183]: why do you plan to remove this?

File: GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java [hunk 10]
```diff
@@ -199,6 +208,30 @@ public void onServiceConnected(ComponentName name, IBinder service) {
...
+                //Start capture if the action is START_CAPTURE
+                if (adbIntent.getAction() != null) {
+                    Log.d(TAG, "Intent action: " + adbIntent.getAction());
+                    if (adbIntent.getAction().equals(
+                            "com.google.gpsdatacapturer.START_CAPTURE")) {
+                        Log.d(TAG, "Start capture via intent");
+                        gpsDataCaptureService.startCapture();
+                    }
+                }
+            }
```
Comment (AdsuleA) [inline GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java]: Please see similar comments on phone activity

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java (hunk 6 above)
Comment (lynnliu655) [reply GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: Yes, I did it on line 228.

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java (hunk 4 above)
Comment (lynnliu655) [reply GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: Yes, I did it on line 228.

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java (hunk 5 above)
Comment (lynnliu655) [reply GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: In current case, the default value is false (locationApiType = LocationManager) if user forgets adding the extra. The user only need to set it be true when they hope to use FusedLocationProvider. Does it make sense?

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/LocationManagerListener.java (hunk 8 above)
Comment (lynnliu655) [reply GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/LocationManagerListener.java]: Yes, we can move it to the service.

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java (hunk 9 above)
Comment (lynnliu655) [reply GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java:183]: The change is I make the `locationApiType` a private field in the service, and call `setLocationApiType()`  in the main activity. Then we don't need to pass`locationApiType` as a parameter to `stopCapture()`. Therefore to stop capture via command line, we don't need extras. ```adb shell am startservice -a com.google.gpsdatacapturer.STOP_CAPTURE ``` is sufficient.

File: GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java (hunk 10 above)
Comment (lynnliu655) [reply GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java]: Will rename it to `launchIntent`

File: GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml (hunk 7 above)
Comment (lynnliu655) [reply GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml:20]: I don't feel we can move it to activity,  as we can only start the activity once, right?

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java (hunk 4 above)
Comment (AdsuleA) [reply GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: Please move the comparison here

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java (hunk 5 above)
Comment (AdsuleA) [reply GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: You should check intent.hasExtra before using getExtras()

File: GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml (hunk 7 above)
Comment (AdsuleA) [reply GpsDataCapturer/sharedlibrary/src/main/AndroidManifest.xml:20]: https://developer.android.com/guide/topics/manifest/activity-element#lmode : Can you try making the activity "singleTask" / "singleInstance"

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java (hunk 9 above)
Comment (AdsuleA) [reply GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java:183]: ack

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 11]
```diff
@@ -68,10 +70,13 @@ protected void onCreate(Bundle savedInstanceState) {
         startAndStopButton = (Button) findViewById(R.id.m_start_stop_button);
         gpsDataTextView = (TextView) findViewById(R.id.m_text_view_gps_data);
         gpsStatusTextView = (TextView) findViewById(R.id.m_text_view_gps_status);
+        satellitesTextView = (TextView) findViewById(R.id.m_text_view_satellites);
 
         //check and request for all necessary permissions
         if (!Utils.hasUserGrantedNecessaryPermissions(this)) {
-            Log.d(TAG, "Permissions required to run this app. Start to request necessary permissions.");
+            Log.d(TAG,
+                    "Permissions required to run this app. Start to request necessary permissions"
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: nit: we can reword sentence. "Missing permissions. Requesting..."

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 12]
```diff
@@ -115,6 +123,24 @@ protected void onCreate(Bundle savedInstanceState) {
         });
     }
 
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+
+        if (intent.getAction() != null) {
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]:
```suggestion
       String action = intent.getAction();
        if (action != null) {

```

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 13]
```diff
@@ -115,6 +123,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            Log.d(TAG, "Intent action: " + intent.getAction());
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]:
```suggestion
           Log.d(TAG, "Intent action: " + action);
```

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 14]
```diff
@@ -115,6 +123,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            //Stop capture if it's stop intent
+            if (intent.getAction().equals(
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]:
```suggestion
            if (action.equals(
```

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 15]
```diff
@@ -44,9 +43,12 @@
     private Button startAndStopButton;
     private TextView gpsDataTextView;
     private TextView gpsStatusTextView;
+    private TextView satellitesTextView;
     private GpsInfoViewModel gpsInfoViewModel;
     private boolean gpsCaptureStopped;
 
+    private Intent launchIntent;
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]:
```suggestion
    private Intent mLaunchIntent;
```

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 16]
```diff
@@ -196,6 +222,17 @@ public void onServiceConnected(ComponentName name, IBinder service) {
             //get the gpsDataCaptureService
             gpsDataCaptureService =
                     ((GpsDataCaptureService.GpsDataCaptureBinder) service).getService();
+
+            if (launchIntent.getAction() != null) {
+                //start capture via launch intent
```
Comment (AdsuleA) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java:229]: please add a check to compare action with START_CAPTURE string

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java [hunk 17]
```diff
@@ -95,4 +92,34 @@ private void onRequestPermissionsResult(int requestCode, String[] permissions,
...
+            Log.d(TAG, "fused_location_type: " + type_from_intent);
+            if (type_from_intent) {
+                locationApiType = LocationApiType.FUSEDLOCATIONPROVIDERCLIENT;
+            }
+            Log.d(TAG, "LocationApiType: " + locationApiType);
+
+            gpsDataCaptureService.setLocationApiType(locationApiType);
+        }
+        //Start capture via intent
+        Log.d(TAG, "Intent action: " + intent.getAction());
```
Comment (AdsuleA) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java]: please move this block to beginning of the method.

File: GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java [hunk 18]
```diff
@@ -116,11 +126,32 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        if (intent.getAction() != null) {
+            Log.d(TAG, "Intent action: " + intent.getAction());
+            //Stop capture if it's stop intent
+            if (intent.getAction().equals(
+                    "com.google.gpsdatacapturer.STOP_CAPTURE")) {
+                Log.d(TAG, "Stop capture via intent onNewIntent");
+                gpsDataCaptureService.stopCapture();
+            } else if (intent.getAction().equals(
+                    "com.google.gpsdatacapturer.START_CAPTURE")) {
+                Utils.startCaptureViaIntent(gpsDataCaptureService, intent, locationApiType);
```
Comment (AdsuleA) [inline GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java]: Please check if the service is bound asking the service to start capture.

Comment (AdsuleA) [inline GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java]: shouldn't locationApiType be extracted from intent?

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java [hunk 19]
```diff
@@ -95,4 +92,34 @@ private void onRequestPermissionsResult(int requestCode, String[] permissions,
     public static boolean isGpsEnabled(LocationManager locationManager) {
         return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
     }
+
+    /**
+     * Start capture via intent
+     */
+    public static void startCaptureViaIntent(GpsDataCaptureService gpsDataCaptureService,
```
Comment (AdsuleA) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java:99]: please add a check for null value of gpsDataCaptureService

File: GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java [hunk 20]
```diff
@@ -45,15 +46,19 @@
     private Button startAndStopButton;
     private TextView gpsDataTextView;
     private TextView gpsStatusTextView;
+    private TextView satellitesTextView;
     private GpsInfoViewModel gpsInfoViewModel;
     private boolean gpsCaptureStopped;
 
+    private Intent launchIntent;
```
Comment (AdsuleA) [inline GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/WearGpsMainActivity.java]:
```suggestion
    private Intent mLaunchIntent;
```

File: GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java [hunk 21]
```diff
@@ -263,6 +309,7 @@ private void hideGpsDataAndStatusTextView() {
     private void resetRadioGroup() {
         apiRadioGroup.check(R.id.m_radio_button_LM);
         apiRadioGroup.setVisibility(View.VISIBLE);
+        locationApiType = LocationApiType.LOCATIONMANAGER;
```
Comment (andriynab) [inline GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java]: do we need to reset locationApiType here?

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java [hunk 22]
```diff
@@ -26,15 +26,12 @@ public void onLocationResult(LocationResult locationResult) {
...
         }
     }
-    @Override
-    public void onLocationAvailability(LocationAvailability var1) {
-    }
 
     @Override
-    public void onGpsStatusChanged(int event) {
-        gpsDataCaptureService.onGpsStatusChanged(event);
+    public void onLocationAvailability(LocationAvailability var1) {
```
Comment (andriynab) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java]: is this override needed?

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/model/GpsInfoViewModel.java [hunk 23]
```diff
@@ -52,6 +54,19 @@ public void setGpsStatusMutableLiveData(int event) {
...
+     * Set the satellietesUsedInFix value
+     * @param  satellites the number of satellites used in fix passed from gps status listener
+     */
+    public void setSatellitesUsedInFix(int satellites){
+        if(satellites == -1){
+            Log.d(TAG, "setSatellitesUsedInFix: UNKNOWN");
+            satellitesUsedInFix.setValue("UNKNOWN");
+        }
+        Log.d(TAG, "setSatellitesUsedInFix");
+        satellitesUsedInFix.setValue("Satellites: " + String.valueOf(satellites));
```
Comment (andriynab) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/model/GpsInfoViewModel.java:67]: this text is part of presentation, right?
preferably should be coming from resource. For example for localization.

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java [hunk 24]
```diff
@@ -231,26 +174,73 @@ public void onLocationChanged(Location location) {
...
      */
     public void onGpsStatusChanged(int event) {
-        //set gps status in the view model
+        int satellitesUsedInFix = 0;
+        if (event == GpsStatus.GPS_EVENT_SATELLITE_STATUS) {
+            //Get Gps Status
+            @SuppressLint("MissingPermission")
+            GpsStatus status = locationManager.getGpsStatus(null);
+
+            assert status != null;
```
Comment (andriynab) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/service/GpsDataCaptureService.java]: if this is important, I think assert should be always checked. There are utils for checking and throwing asserts like this that doesn't require special runtime flags.

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java [hunk 25]
```diff
@@ -95,4 +92,35 @@ private void onRequestPermissionsResult(int requestCode, String[] permissions,
...
+        Log.d(TAG, "Intent action: " + action);
+
+        if(gpsDataCaptureService != null) {
+            //Extra the location api type
+            LocationApiType locationApiType = LocationApiType.LOCATIONMANAGER;
+            if (intent.hasExtra("fused_location_type")) {
+                boolean type_from_intent = intent.getBooleanExtra("fused_location_type", false);
+                Log.d(TAG, "fused_location_type: " + type_from_intent);
+
+                if (type_from_intent) {
```
Comment (andriynab) [inline GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/utils/Utils.java]: type_from_intent : snake  names are not used in Java. Also if boolean it should be in predicate form, isA or hasA. 
will be more readable if type_from_intent is not boolean but enum:
if (typeFromIntent == FUSED_API) ...

but if working with boolean name should suggest what property we are checking
if (isTypeFromIntentFused) { ...

**Review by andriynab — APPROVED** [review]

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java (hunk 22 above)
Comment (lynnliu655) [reply GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/FusedLocationProviderListener.java]: Removed.

File: GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/model/GpsInfoViewModel.java (hunk 23 above)
Comment (lynnliu655) [reply GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/model/GpsInfoViewModel.java:67]: Yes, it's part of the presentation. I was slacking off at "drawing"  too many text view in the layout and wrapped "Satellites: " text with the live data in one text view.

**Review by AdsuleA — APPROVED** [review]

**Pull request merged**